File: Arrays_&_Hashing/Group_Anagrams/1_BruteForce.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        int n = strs.size();
        vector<vector<string>> res;
        vector<bool> used(n,false);
        for(int i=0;i<n;++i){
            if(used[i]) continue;
            vector<string> group;
            group.push_back(strs[i]);
            used[i]=true;
            for(int j=i+1;j<n;++j){
                if(!used[j] && isAnagram(strs[i], strs[j])){
                    group.push_back(strs[j]);
                    used[j]=true;
                }
            }
            res.push_back(move(group));
        }
        return res;
    }
private:
    bool isAnagram(const string& a, const string& b){
        if(a.size()!=b.size()) return false;
        int cnt[26]={0};
        for(char c:a) cnt[c-'a']++;
        for(char c:b) cnt[c-'a']--;
        for(int x:cnt) if(x!=0) return false;
        return true;
    }
};

File: Arrays_&_Hashing/Group_Anagrams/1_BruteForce.java
import java.util.*;
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        int n = strs.length;
        boolean[] used = new boolean[n];
        List<List<String>> res = new ArrayList<>();
        for(int i=0;i<n;i++){
            if(used[i]) continue;
            List<String> group = new ArrayList<>();
            group.add(strs[i]);
            used[i]=true;
            for(int j=i+1;j<n;j++){
                if(!used[j] && isAnagram(strs[i], strs[j])){
                    group.add(strs[j]);
                    used[j]=true;
                }
            }
            res.add(group);
        }
        return res;
    }
    private boolean isAnagram(String a, String b){
        if(a.length()!=b.length()) return false;
        int[] cnt = new int[26];
        for(char c: a.toCharArray()) cnt[c-'a']++;
        for(char c: b.toCharArray()) cnt[c-'a']--;
        for(int v: cnt) if(v!=0) return false;
        return true;
    }
}
